### PR TITLE
Display info message for missing whole sections of eCR Viewer

### LINKED ISSUE #2112: Display informational message to users when whole sections of the eCR are missing
## Action Requested

If we don't have any data available at all for any of the main sections of the eCR Viewer:

- Patient Info
- Encounter Info
- Clinical Info
- Lab Info
- eCR Metadata
- Unavailable Info

we should have an informational message below the header for that section to explain why there is no data there.

## Acceptance Criteria

If there is missing info for any of the below sections, then the corresponding informational message shows up below that header, formatted like in the designs below:

- [x] if there is no Patient Info (note this is highly unlikely) --> "No patient information was found in this eCR."
- [x] if there is no Encounter Info (also highly unlikely) --> "No encounter information was found in this eCR."
- [x] if there is no Clinical Info --> "No clinical information was found in this eCR."
- [x] if there is no Lab Info --> "No lab information was found in this eCR."
- [x] if there is no eCR Metadata (also highly unlikely) --> "No eCR metadata was found in this eCR."
- [x] if there is no Unavailable Info (very very unlikely) --> "All possible information was found in this eCR."

## Testing

???

## Designs

[Link to Figma Designs](https://www.figma.com/design/tXWUeV3Kjehd7VNqHojZYq/eCR-Viewer?node-id=4297-25491&t=Qjo2ItVdsCCiRRg5-1)

## Background & Context

Right now, because we don't display any information about why a section is blank, it's a bit confusing to understand why it's blank. The user may wonder if there is a bug or error that's preventing info from being shown. By adding this message, users can feel confident that there is just no information available for that section, and the Viewer is working correctly.

### PR DESCRIPTION
**Pull request by angelathe**
# PULL REQUEST

## Summary
- Displays information messages for whole sections that are missing in the eCR Viewer
- Add snapshot test

## Related Issue
Fixes #2112 

## Additional Information
See comments below for a few notes.

## Acceptance Criteria
If there is missing info for any of the below sections, then the corresponding informational message shows up below that header, formatted like in the designs below:

- [ ] if there is no Patient Info (note this is highly unlikely) --> "No patient information was found in this eCR."
- [ ] if there is no Encounter Info (also highly unlikely) --> "No encounter information was found in this eCR."
- [ ] if there is no Clinical Info --> "No clinical information was found in this eCR."
- [ ] if there is no Lab Info --> "No lab information was found in this eCR."
- [ ] if there is no eCR Metadata (also highly unlikely) --> "No eCR metadata was found in this eCR."
- [ ] if there is no Unavailable Info (very very unlikely) --> "All possible information was found in this eCR."

## Screenshot
(It's hard to test this when most sections are never empty)
Screenshot 2024-08-13 at 23 19 01

## Testing
1. To view a single section that is missing, view the "Lab Info" section for these sample eCRs: [Link 1](http://localhost:3000/view-data?id=a3266d94-ce19-11ec-81e1-8465ad080d41#lab-info) [Link 2](http://localhost:3000/view-data?id=016accab-d6bf-11ec-81e1-8465ad080d41#lab-info)
2. To view the info messages for all the sections except for the Patient Info/Unavailable Info section, add the following bundle through a db client and follow it's link.
```
{
  "type": "batch",
  "entry": [
  ],
  "resourceType": "Bundle"
}
```
3. To view the info message for the Unavailable Info section, DM me for the code to add to `AccordionContent.tsx`

[//]: # (PR title: Remember to name your PR descriptively!)

### CODE REVIEW

File: containers/ecr-viewer/src/app/tests/components/AccordionContent.test.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    expect(await axe(container)).toHaveNoViolations();
+    container.querySelectorAll("[id], [aria-describedby]").forEach((el) => {
+      el.removeAttribute("id");
+      el.removeAttribute("aria-describedby");
+    });
+    expect(container).toMatchSnapshot();
+
+    // expect(
+    //   screen.getByText("No patient information was found in this eCR."),
+    // ).toBeInTheDocument();
```
Comment (angelathe) [inline containers/ecr-viewer/src/app/tests/components/AccordionContent.test.tsx:31]: Note: Commented this out because there's currently a bug where even when there's no patient info, certain functions will make Patient Name/Vital Signs/Contact non-empty. Will make sure this gets uncommented in the issue's (to be created soon) acceptance criteria.

File: containers/ecr-viewer/src/app/tests/components/AccordionContent.test.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import { render, screen } from "@testing-library/react";
+import { axe } from "jest-axe";
+import { Bundle } from "fhir/r4";
+import { loadYamlConfig } from "@/app/api/utils";
+import AccordionContent from "@/app/view-data/components/AccordionContent";
+
+const mappings = loadYamlConfig();
+
+describe("Snapshot test for Accordion Content", () => {
+  it("Given no data, info message for empty sections should appear", async () => {
```
Comment (angelathe) [inline containers/ecr-viewer/src/app/tests/components/AccordionContent.test.tsx:11]: I also wanted to add a test that was "Given all the data, info message for empty Unavailable Info section should appear," but I think that would've required building a FHIR bundle complete with all info. I ended up not doing it because there are pieces of info I haven't seen in any of the sample data.

Let me know if we think this is a priority, though, this could be made into a separate ticket!

**Comment by angelathe** [comment]
@sarahtress FYI, the design review script may not be useful for testing how each section looks when empty. You'll be able to view empty "Lab Info" sections in these sample eCRs ([Link 1](http://localhost:3000/view-data?id=a3266d94-ce19-11ec-81e1-8465ad080d41#lab-info) [Link 2](http://localhost:3000/view-data?id=016accab-d6bf-11ec-81e1-8465ad080d41#lab-info)), but if you want to see the other sections, let me know and we can pair!

p.s. attached a screenshot of what they would all look like for good measure

**Comment by sarahtress** [comment]
@angelathe this looks good to me - thanks!

**Review by BobanL — APPROVED** [review]
👍
